beta-prompts/beta_prompts/templating_framework.py

```
@@ -0,0 +1,1029 @@
+    - Context-aware adaptation
+    - Variable validation and substitution
+    - Performance optimization
+    - Template versioning and management
+    """
+
+    def __init__(self, registry_path: str = "./template_registry"):
+        self.registry = TemplateRegistry(registry_path)
+        self.context_engine = ContextEngine()
+
+        # Initialize Jinja2 environment
+        self.jinja_env = Environment(
```

Ruff (0.12.2)

By default, jinja2 sets `autoescape` to `False`. Consider using `autoescape=True` or the `select_autoescape` function to mitigate XSS vulnerabilities.

(S701)